### PR TITLE
Documentation build system improvement (1/3)

### PR DESCRIPTION
**Pull request by natanlao**
This is pull request 1 of 3:

- [x] Build system improvements (vaguely specified in #203) made to `develop`
- [ ] Build system improvements made to `gh-pages`
- [ ] Documentation of above changes once the above two PRs are merged

This PR implements the following changes:
- Reduces footprint of Swagger UI documentation by framing a central, vendored copy of swagger-ui. This reduces per-branch overhead from ~8.0M to ~8.0K, which doesn't seem like that much, but that I think will scale more easily and ideally make builds a little faster.
- Removed some lingering artifacts and tweaked `.gitignore` to keep them out of the process
- Removed a good amount of the Node/npm calls

### CODE REVIEW

**Comment by natanlao** [comment]
Also, as I mentioned in the earlier PR, this build system will not function for feature branches until `develop` is merged into `master` @briandoconnor

**Comment by briandoconnor** [comment]
Hey @jaeddy do you have any bandwidth to look at this before next Monday's call?

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,18 @@ SMARTAPI_PATH = openapi/data_repository_service.smartapi.yaml
 SMARTAPI_PART_PATH = openapi/data_repository_service.smartapi.yaml.part
 SWAGGER2OPENAPI_PATH = swagger2openapi
 
+REPO_URL ?= "https://github.com/$(TRAVIS_REPO_SLUG)"
+BRANCH_NAME ?= $(echo "$(TRAVIS_BRANCH)" | awk '{print tolower($0)}')
+DOCS_STAGING_PATH = deploy
+
+# Make docs available at / for master and at /preview/{branch} for all others.
+# We don't need to check if `branch == gh-pages` since Travis will skip it by default.
+ifeq "$(BRANCH_NAME)" "master"
```
Comment (jaeddy) [inline Makefile]: For WES, we had agreed to make `develop` the main "source of truth" for docs, status badges, and GitHub Pages content. While `master` would be the default branch visible to new visitors (post 1.0), it mostly functions as a passthrough for releases.

I don't think this is worth addressing here, but making a note that we should try to sync up on branch usage, PR procedure, etc. across repos (whether it's following "HubFlow" or something else).

**Comment by delagoya** [comment]
+1

**Comment by denis-yuen** [comment]
FYI @garyluu for https://github.com/ga4gh/dockstore/issues/1912

**Comment by natanlao** [comment]
I have updated documentation that reflects the changes made in this PR. I'll add it to the repo once this is merged, just in case there are any kinks in my PR that I haven't caught yet.

**Comment by dglazer** [comment]
@natanlao , I just merged #231 (after getting @briandoconnor 's approval). You indicated in your approval that you'd need to make some changes to this PR as a result -- flagging so we don't lose track.

**Comment by jaeddy** [comment]
@briandoconnor / @natanlao — was there any reason that this PR was never merged/closed?

**Pull request closed without merging**